schema/pgsql/upgrades/017.sql

```diff
@@ -0,0 +1,21 @@
+ALTER TABLE contact ADD COLUMN default_channel_id bigint REFERENCES channel(id);
+
+UPDATE contact SET default_channel_id = (SELECT id FROM channel WHERE channel.type = 'rocketchat' LIMIT 1) WHERE default_channel = 'rocketchat';
+UPDATE contact SET default_channel_id = (SELECT id FROM channel WHERE channel.type = 'email' LIMIT 1) WHERE default_channel = 'email';
```

Comment:
You shouldn't have to even give the channel types yourself. It should be perfectly fine to reference columns from the outer query in the subquery, so it should be possible to just do:
```sql
UPDATE contact
SET default_channel_id = (SELECT id FROM channel WHERE channel.type = contact.default_channel LIMIT 1);
```